libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -704,7 +711,52 @@ int32_t ESP8266WiFiSTAClass::RSSI(void) {
     return wifi_station_get_rssi();
 }
 
+void ESP8266WiFiSTAClass::stationKeepAliveNow ()
+{
+    if (_keepStationAliveUs > 0)
+    {
+        for (netif* interface = netif_list; interface != nullptr; interface = interface->next)
+             if (
+                   (interface->flags & NETIF_FLAG_LINK_UP)
+                && (interface->flags & NETIF_FLAG_UP)
+#if LWIP_VERSION_MAJOR == 1
+                && interface == eagle_lwip_getif(STATION_IF) /* lwip1 does not set num properly */
+                && (!ip_addr_isany(&interface->ip_addr))
+#else
+                && interface->num == STATION_IF
+                && (!ip4_addr_isany_val(*netif_ip4_addr(interface)))
+#endif
+            )
+            {
+                etharp_gratuitous(interface);
+                break;
+            }
+
+            // re/schedule with a possibly updated interval
+            schedule_recurrent_function_us([&]()
+            {
+                stationKeepAliveNow();
+                return false;
+            }, _keepStationAliveUs);
+    }
+    else
+        // mark disabled (0) if user cancelled it (<0)
+        _keepStationAliveUs = 0;
+}
 
+void ESP8266WiFiSTAClass::stationKeepAliveSetIntervalMs (int ms)
+{
+    bool wasEnabled = (_keepStationAliveUs > 0);
+    int us = ms * 1000;
+    if (wasEnabled)
+        _keepStationAliveUs = ms <= 0? /*disable*/-1: /*update*/us;

```

Comment:
Maybe you should also check for negative values of us (or ms > (max int  / 1000) )